bobtimus/src/bin/fake_bobtimus.rs

```diff
@@ -59,9 +61,24 @@ async fn main() -> Result<()> {
                     }
                 });
 
-            warp::serve(routes.or(faucet).with(cors))
-                .run(([127, 0, 0, 1], api_port))
-                .await;
+            match (tls_certificate, tls_private_key) {
+                (Some(certificate), Some(private_key)) => {
+                    warp::serve(routes.or(faucet).with(cors))
+                        .tls()
+                        .cert_path(certificate)
+                        .key_path(private_key)
+                        .run(listen_address)
+                        .await;
+                }
```

Comment:
Does this serve http and https or is this now an or? 

We should serve http and https at the same time. E.g. if the provider wants to serve HTTPS and Tor. Tor needs to go against http only.

Reply:
Interestingly enough Tor + HTTPS [can actually be set up](https://community.torproject.org/onion-services/advanced/https/) - so I am tempted to say that, if a provider has a certificate he might as well just use that for the onion setup as well. 
I have, however, not tested this yet. Additionally I am not sure about the implications. Do you think it would make sense to validate that such a setup works (by running it) or would you just enforce serving it on both `http` and `https` if user configures it?
I think we can provide a separate listening address for https as well and enable serving both. As long as all components that keep state are fine with that (e.g. database being accessed from two different components) that should be fine. Are we using transactions to write into the DB already?

Reply:
After reading through the article I posted above one more time I think I take back what I said. We *will* have to serve on `http` and `https` - there are too many unknowns otherwise. I will see to change this accordingly.

Reply:
Just to clarify things: 

Assuming bobtimus has the domain `bob.comit.network` and has a certificate for the this domain he can run `https://bob.comit.network`. 
If he creates a Tor hidden service his service would be reachable under `thisisarandomononionaddress.onion`. 

Note that the certificate for `bob.comit.network` is not valid for `thisisarandomononionaddress.onion`. 

So if bobtimus would only accept https requests a request to **https**`://thisisarandomononionaddress.onion` will return an invalid certificate and a browser warning. 

In theory, bobtimus could get a certificate for `thisisarandomononionaddress.onion` but this is really hard (e.g. letsencrypt does not support this.) 

Hence, best practice is to disable https when provided hidden service which are reachable via an onion address.

Reply:
yeah, I also noticed that - he would need *two* certificates (with the second one being the one for the onion address...) and the is really not what is wanted. Already changed, I did the changes as `!fixup` in case you want to see them in isolation. Will squash that later :)